### DISCUSSION TITLE
Production deploy based on yarn static:build static files

### DISCUSSION THREAD

**Discussion by mauricio-raio** [BearStudio/start-ui-web#236] [Q&A] [answered]
Hi, I'm trying to deploy the static site version in production, but it has the refresh page issue described in the stackoverflow thread in the link below. I've created the rewrite rule from `/app/*` to `app.html`, but it seems not to be enough. 

Has anyone have success in making this static site deploy and have site refresh working? 

Problem description: https://stackoverflow.com/questions/27928372/react-router-urls-dont-work-when-refreshing-or-writing-manually

P.S. I have tried some of the suggestion from stackoverflow, without success.

**Comment by Rileran** [comment]
Where did you configure the rewrite from `/app/*` to `app.html` on your production server ? The rewrite can't be configured directly inside the project, but should be configured on the server serving the static files.

If you are using some kind of apache server to statically deploy your app, you should add a file `public/.htaccess` in the source code of your nextjs projet. It will contain the rewrite rule.

You should have something like the following: 

```
# public/.htaccess
Options -MultiViews
RewriteEngine On

# Rewrite /app/* to app.html
RewriteCond %{REQUEST_FILENAME} !-f
RewriteRule "^app/" "app.html" [QSA,L]
```

Then this configuration file will be available in the apache server, and the rewriting will be done correctly.

**Reply by mauricio-raio** [reply]
Hi, @Rileran . Thanks for the reply. 

I'm deploying on render.com and they have a Redirects/Rewrites configuration session where I just have to fill the fields **source** and **destination**. 

I've filled source with "/app/*" and destination with "app.html"  for "Rewrite" action. I know that it takes some effect because, without this configuration, no page is rendered. But page refresh still not working.

I'll have a try with a local apache server, before move on to production.

Thank you.

**Comment by mauricio-raio** [comment, marked as answer]
Hi, @Rileran. After trying with a local apache server, I discovered that in the .htaccess example I should be using index.html instead of app.html, like below:
RewriteRule "^app/" "index.html" [QSA,L]

The `yarn static:build` won't generate any app.html file and Apache was rewriting the url to nowhere. Just fixed the file name and it worked as expected on Apache. 

On render.com I just used the rewrite configuration below:

Source: **/app/***
Destination: **/index.html**

Everything is working as expected.

Thanks

**Reply by Rileran** [reply]
Hi, @mauricio-raio !

Happy to see that you've managed to successfully deploy your app !

Your case will help us improve our documentation on deploying start-ui. Thanks !
